stacks-core/src/utils.rs

```diff
@@ -111,6 +112,36 @@ impl From<StacksAddress> for PrincipalData {
 	}
 }
 
+impl TryFrom<String> for PrincipalData {
+	type Error = StacksError;
+
+	fn try_from(value: String) -> Result<Self, Self::Error> {
+		let parts: Vec<&str> = value.split('.').collect();
+
+		match parts.len() {
+			1 => {
+				let address = StacksAddress::try_from(parts[0])?;
+				Ok(Self::Standard(StandardPrincipalData::from(address)))
+			}
+			2 => {
+				let address = StacksAddress::try_from(parts[0])?;
+				let contract_name =
+					ContractName::new(parts[1]).map_err(|_err| {
+						StacksError::InvalidData("Invalid contract name")
+					})?;
+
+				Ok(Self::Contract(
+					StandardPrincipalData::from(address),
+					contract_name,
+				))
+			}
+			_ => Err(StacksError::InvalidData(
+				"Principal data may contain at most 1 dot character",
```

Comment:
What value can't be split into parts?

Reply:
Eg. "SP1234.hello.world"

Reply:
What I mean is that the error does not communicate what the `value` being parsed is. I expect my errors to tell me what input"SP1234.hello.world" and why it failed.